tap_xactly/helpers.py

```diff
@@ -0,0 +1,19 @@
+from singer import metadata
+from tap_xactly.streams import STREAMS
+from tap_xactly.client import XactlyClient
+
+
+def get_stream_object(stream, state, config):
+    return STREAMS[stream.tap_stream_id](get_client(config), state)
```

Comment:
Every time this method is called, which is one per stream as implemented in `sync.py`, it creates a new instance of `XactlyClient` due to `get_client()`. We should really only instantiate one instance of `XactlyClient` and pass it to each stream.